src/python/library/tritonclient/grpc/aio/__init__.py

```diff
@@ -0,0 +1,762 @@
+            if self._verbose:
+                if name != "":
+                    print(
+                        "Unregistered cuda shared memory with name '{}'".format(
+                            name))
+                else:
+                    print("Unregistered all cuda shared memory regions")
+        except grpc.RpcError as rpc_error:
+            raise_error_grpc(rpc_error)
+
+    async def infer(self,
+                    model_name,
+                    inputs,
+                    model_version="",
+                    outputs=None,
+                    request_id="",
+                    sequence_id=0,
+                    sequence_start=False,
+                    sequence_end=False,
+                    priority=0,
+                    timeout=None,
+                    client_timeout=None,
+                    headers=None,
+                    compression_algorithm=None):
+        """Refer to tritonclient.grpc.InferenceServerClient
+
+        """
+
+        if headers is not None:
+            metadata = headers.items()
+        else:
+            metadata = ()
+
+        if type(model_version) != str:
+            raise_error("model version must be a string")
+
+        request = _get_inference_request(model_name=model_name,
+                                         inputs=inputs,
+                                         model_version=model_version,
+                                         request_id=request_id,
+                                         outputs=outputs,
+                                         sequence_id=sequence_id,
+                                         sequence_start=sequence_start,
+                                         sequence_end=sequence_end,
+                                         priority=priority,
+                                         timeout=timeout)
+        if self._verbose:
+            print("infer, metadata {}\n{}".format(metadata, request))
+
+        try:
+            response = await self._client_stub.ModelInfer(
+                request=request,
+                metadata=metadata,
+                timeout=client_timeout,
+                compression=_grpc_compression_type(compression_algorithm))
+            if self._verbose:
+                print(response)
+            result = InferResult(response)
+            return result
+        except grpc.RpcError as rpc_error:
+            raise_error_grpc(rpc_error)
+
+    async def stream_infer(self,
+                           inputs_iterator,
+                           stream_timeout=None,
+                           headers=None,
+                           compression_algorithm=None):
+        """Runs an asynchronous inference over gRPC bi-directional streaming
+        API.
+
+        Parameters
+        ----------
+        inputs_iterator : async_generator
+            Async iterator that yields a dict(s) consists of the input 
+            parameters to the async_stream_infer function defined in 
```

Comment:
async_stream_infer -> stream_infer

Reply:
Actually, `async_stream_infer` is correct.

It is because the `inputs_iterator` is acting like the non-async `async_stream_infer` function. In other words, this async `stream_infer` function replaces the non-async `start_stream` and `stop_stream` function in the async implementation, and the `inputs_iterator` is there to yield the input(s) previously passed into the non-async `async_stream_infer` function. Therefore, the `inputs_iterator` accepts exactly the parameters into the non-async `async_stream_infer` function, in a dictionary. It is similar to the non-async `stream_infer` parameters but without the `callback`, `client_timeout`, `headers` and `compression_algorithm` parameters, as they are passed as parameters into this async `stream_infer` function. Thus, `async_stream_infer` is more appropriate here.

(Note: **non-async** is referring to functions in `tritonclient.grpc.InferenceServerClient` and **async** is referring to those in `tritonclient.grpc.aio.InferenceServerClient`)

Maybe I just list out the entire `inputs_iterator`'s yield instead?

Reply:
You are right. This makes sense.